sound/soc/sof/pcm.c

```diff
@@ -229,6 +231,27 @@ static int sof_pcm_trigger(struct snd_pcm_substream *substream, int cmd)
 		stream.hdr.cmd |= SOF_IPC_STREAM_TRIG_PAUSE;
 		break;
 	case SNDRV_PCM_TRIGGER_RESUME:
+
+		/* resume stream */
+		host_posn = spcm->stream[substream->stream].posn.host_posn;
+		host = bytes_to_frames(substream->runtime, host_posn);
+		dev_dbg(sdev->dev,
+			"PCM: resume stream %d dir %d DMA position %lu\n",
+			spcm->pcm.pcm_id, substream->stream, host);
+
+		/* set hw_params */
+		ret = sof_pcm_hw_params(substream,
+					&spcm->params[substream->stream]);
+		if (ret < 0) {
+			dev_err(sdev->dev,
+				"error: set pcm hw_params after resume\n");
+			return ret;
+		}
+
+		/* trigger stream */
+		stream.hdr.cmd |= SOF_IPC_STREAM_TRIG_START;
```

Comment:
Where do we send the IPC start message if we set cmd here ?

Reply:
@lgirdwood It is sent right outside the switch/case block where we set the cmd.